homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -285,11 +282,28 @@ def _calculate_new_probability(self):
 
         for obs in self.current_observations.values():
             if obs is not None:
-                prior = update_probability(
-                    prior,
-                    obs["prob_given_true"],
-                    obs.get("prob_given_false", 1 - obs["prob_given_true"]),
-                )
+                if obs["observation"] is True:
+                    prior = update_probability(
+                        prior,
+                        obs["prob_given_true"],
+                        obs.get("prob_given_false", 1 - obs["prob_given_true"]),
+                    )
+                elif obs["observation"] is False:
+                    prior = update_probability(
+                        prior,
+                        1 - obs["prob_given_true"],
+                        1 - obs.get("prob_given_false", 1 - obs["prob_given_true"]),
+                    )
+                elif obs["observation"] is None:
+                    if obs["entity_id"] is not None:
+                        _LOGGER.info(
+                            "Observation for entity '%s' returned None, it will not be used for Bayesian updating",
+                            obs["entity_id"],
+                        )
```

Comment:
This looks like it's meant for debugging? If it is, please lower severity to `debug`.

Reply:
Hmm, could be... Although this will only print when a sensor variable returns some form of `None` which in many cases will be an unexpected input and may cause unexpected behaviour for a user - namely that one of their sensors is unavailable. If you feel this is a debugging only step then happy to change the log level.

Reply:
Fixed in 8baf93ee21df1e1024a8e99922ef3f793a183069